veloxms-core/src/main/scala/edu/berkeley/veloxms/VeloxApplication.scala

```diff
@@ -92,9 +94,53 @@ class VeloxApplication extends Application[VeloxConfiguration] with Logging {
     jsonMapper.readValue(json, classOf[Seq[String]])
   }
 
+  def createModel(name: String,
+                  sparkContext: SparkContext,
+                  etcdClient: EtcdClient,
+                  broadcastProvider: BroadcastProvider,
+                  sparkDataLocation: String,
+                  partitionMap: Seq[String],
+                  env: Environment,
+                  hostname: String
+                  ): Unit = {
+
+    val key = s"$configEtcdPath/models/$name"
+    val json = etcdClient.getValue(key)
+    val jsonTree = jsonMapper.readTree(json)
+    require(jsonTree.has("modelType"))
+    require(jsonTree.has("onlineUpdateDelayInMillis"))
+    require(jsonTree.has("batchRetrainDelayInMillis"))
+    val modelType = jsonTree.get("modelType").asText
+    val onlineUpdateDelay = jsonTree.get("onlineUpdateDelayInMillis").asLong
+    val batchRetrainDelay = jsonTree.get("batchRetrainDelayInMillis").asLong
+    val modelSpecificConfig: Option[JsonNode] = if (jsonTree.has("config")) {
+      Some(jsonTree.get("config"))
+    } else {
+      None
+    }
+    val mirror = ru.runtimeMirror(getClass.getClassLoader)
```

Comment:
Oh this must have been a pain to figure out. You add some comments as to what this is doing (so we can update in case of a reflections api change) & note what assumptions about the model implementations are being made here? (e.g. no-op or one-op constructor)